### ISSUE TITLE
Display the policy nicely (like in installer) in the settings

### ISSUE THREAD

**Issue by nostitos** [wizardsardine/liana#620]
Such a great UI to create the wallet. 

Why not use the same interface to visualize what are the different recovery fallbacks are after creation?

**Comment by darosior** [comment]
Because:
- We show you what needs to be backed up (and checked against the signing device, ideally from the backup)
- Showing you the same UI as the previous screen would be redundant? I think it's pretty clear from this first step what fallbacks you've got and showing it again would basically make us duplicate this screen?

**Comment by kloaec** [comment]
I believe @nostitos is talking about the wallet itself, not the installer.
We could potentially "show" the installer view of paths in the settings or something.

The way we do this now is in the Recovery screen, when UTXOs are ready to recover, we display all the valid paths to spend.
But it's only visible once a UTXO timelock lapsed, maybe we could show it at the Descriptor screen too?

Something like:
Descriptor: `display descriptor here`
Visualisation:
Primary path: 2 of (_Ledger 1_) (_Bitbox 1_)
Recovery path 1: **timelock 10000 blocks**, 2 of (_Ledger 1_) (_Bitbox 1_) (_backup key_)
Recovery path 2: **timelock 50000 blocks**, (_Last Key_)

**Comment by nondiremanuel** [comment]
I had a similar request in a different issue, but since it's already here I will only add my comment and close the other issue: I believe it could be a nice improvement to show the spending policy in some more friendly way, either graphical or otherwise. The use case that I have in mind is: average Bob creates a new Liana wallet and before starting receiving his funds wants to be sure that he set the correct spending policy. If he doesn't have a way to do it, he could be refrained from using Liana.

In my mind, ideally we could also add such view while the wallet creation is ongoing (showing the paths populate at each step), but this is probably part of the ongoing review of the onboarding process.

**Comment by pythcoiner** [comment]
where do you this should be the better place put put this? in settings menu? or somewhere else?

**Comment by pythcoiner** [comment]
quite related to #516, i think we can at least reuse the same widget/building block

**Comment by nondiremanuel** [comment]
Waiting for the onboarding UX review to see if we can reuse something also in the settings

**Comment by pythcoiner** [comment]
> Waiting for the onboarding UX review to see if we can reuse something also in the settings

yes, just comment as a reminder

**Comment by nondiremanuel** [comment]
> > Waiting for the onboarding UX review to see if we can reuse something also in the settings
> 
> yes, just comment as a reminder

My comment was not related to yours, sorry for the lack of clarity. I was just updating the issue to keep everyone informed about its status :)

**Comment by nondiremanuel** [comment]
After yesterday's call, let's divide the work in two parts:
1. Display the policy form in written form, with the possibility to copy it, in the settings
2. Possibly add a graphical form reusing something from the new onboarding

Since the second point requires the design finalization (#1147), let's postpone it while focusing on the first one as a nice to have for v7

**Comment by nondiremanuel** [comment]
So, for the first point about having the policy displayed in written form, I would aim to have a standard way to explain it in a way that is understandable by everyone. We should mainly explain:
- When a specific set of keys is able to spend the wallet's funds: primary path vs recovery path(s)
   - In case of primary path: "`primary path set of keys` can always spend this wallet’s funds"
   - In case of recovery path: "`recovery path set of keys` can recover this wallet’s funds afer ~`timelock duration` of inactivity"
- Define `primary path set of keys` or the `recovery path set of keys`, by explaining what composes them: single sig vs multisig:
   -  In case of single sig: "1 signature by `key 1 alias`" ...
   - In case of multisig: "`M` signatures out of `N` by `key 1 alias` … and `key N alias`" ...

Notes:
`timelock duration` could be in the format: duration in y, m, d, h, mn (as in other parts of the application) plus the indication of the number of blocks in parenthesis. I would possibly avoid showing the h and mn if the timelock is over 1 year, so maybe we could show the 2 or 3 more relevant units.
`M` and `N` are the parameters defining the multisig quorum.

So in an example of a descriptor with a 2 of 2 multisig composed by "Key 1" and "Key 2" as primary path, a 2 of 3 multisig adding "Key 3" as first recovery path (1 year timelock) and a single sig with "Key 3" as a second recovery path (1 year and 3 months timelock), in my proposal we would show something like:

"Wallet's spending policy:

**2 signatures** out of 2 by **Key 1** and **Key 2** can always spend this wallet's funds. [Primary Path]
**2 signatures** out of 3 by **Key 1**, **Key 2** and **Key 3** can recover this wallet's funds after **~1 year** (52596 blocks) of inactivity. [Recovery Path # 1] 
**1 signature** by **Key 3** can recover this wallet's funds after **~1y 2m 28d** (65535 blocks) of inactivity". [Recovery Path # 2]"

Parts of the wallet in which this would be shown:
- Wallet settings in the wallet info (I would put it as first thing)
- When a participant joins a wallet via email and pastes the received token: as a confirmation step we should show this (together with descriptor to backup?)
- Maybe we can add it in the wallet creation flow as an intermediate step after setting up the descriptor and before backing it up. This would help the user confirm what he just did.

Please let me know if you see any points I'm missing and what you think in general if you'd like to

**Comment by edouardparis** [comment]
by Key1, Key2, do you mean just the fingerprint or the alias or both with an hover ?

**Comment by nondiremanuel** [comment]
> by Key1, Key2, do you mean just the fingerprint or the alias or both with an hover ?

I meant the alias the user gave at wallet creation. In the case, if present, in which they are not available, we could use fingerprints. Do you think we need the fingerprint even when we have aliases too?

**Comment by edouardparis** [comment]
I think alias + fingerprint on hover is the best. Also I think we should change the wording of
> can recover this wallet's funds after ~1y 2m 28d (65535 blocks) of inactivity

for 
can recover funds unmoved for 1y.... (... blocks)

in order to highlight the per utxo policy

**Comment by nondiremanuel** [comment]
ok. I would use "not moved" instead of unmoved though

**Comment by jp1ac4** [comment]
Given that we have a Coins page and the timelocks apply per coin, do you think it would be better to refer to coins rather than funds to be clear that the recovery is on a per-coin basis?

I think all suggested wording options could work, although perhaps we can be more explicit about what we mean by "inactive" and "unmoved", e.g. we could use "spent/refreshed", especially if referring to coins, e.g. "can recover any coins in this wallet that have not been spent or refreshed for 1y...".

Another option could be "coins that are older than 1y...".

**Comment by edouardparis** [comment]
I prefer the shorter sentence, the line is already quite long with all the keys in it

>coins that are older than 1y..

**Comment by edouardparis** [comment]
> When a participant joins a wallet via email and pastes the received token: as a confirmation step we should show this (together with descriptor to backup?)

Can we crate a separate issue for the enhancement of this panel ? It is a little bit out of scope as it as some requirements on the backend like sharing descriptor even if user does not accept yet the invitation and we should discuss it.
